### PR TITLE
Refactor `README.md` and docs

### PR DESCRIPTION
**Pull request by upbqdn**
## Motivation

Prepare for releasing v1.8.

## Solution

- [Move "Known Issues" from README to the book](https://github.com/ZcashFoundation/zebra/commit/50e34a0001b4a1129d29edf63742e30faadd83e9)
- [Remove an old note from the book](https://github.com/ZcashFoundation/zebra/commit/2cd5142fa8265c82e1a306ab89486b1269467ff4)
- [Move optional configs & features to the book](https://github.com/ZcashFoundation/zebra/commit/66f387ad227c9106f8c4abffcb03fac75c647d5d)
- [Refactor the section on installing Zebra](https://github.com/ZcashFoundation/zebra/commit/738337480891d26681ab818e34eb9a363a720ed5)
- [Fix badges in README.md](https://github.com/ZcashFoundation/zebra/commit/d2705ad6214f7aa58c8731f3f99aed3517dfe553)

### Tests

Not needed since this is only a docs refactor.

### PR Author's Checklist

- [x] The PR name will make sense to users.
- [ ] ~~The PR provides a CHANGELOG summary.~~
- [x] The solution is tested.
- [x] The documentation is up to date.
- [x] The PR has a priority label.

### PR Reviewer's Checklist

- [x] The PR Author's checklist is complete.
- [x] The PR resolves the issue.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,67 +121,6 @@ zebrad start
...
-```sh
-cargo install --features="<feature1> <feature2> ..." ...
-```
-
-Our full list of experimental and developer features is in [the API documentation](https://docs.rs/zebrad/latest/zebrad/index.html#zebra-feature-flags).
-
-Some debugging and monitoring features are disabled in release builds to increase
-performance.
-
-## Known Issues
```
Comment (arya2) [inline README.md:176]: Could we leave this section of the README with a link to the troubleshooting page of the book so it's still easy to find?

File: README.md [hunk 2]
```diff
@@ -126,67 +121,6 @@ zebrad start
 See the [Installing Zebra](https://zebra.zfnd.org/user/install.html) and [Running Zebra](https://zebra.zfnd.org/user/run.html)
 sections in the book for more details.
 
-#### Optional Configs & Features
```
Comment (arya2) [inline README.md:129]: Could we leave this section too, with just a link to the "install" page of the book?

**Review by arya2 — DISMISSED** [review]
This looks good, thank you!

File: README.md (hunk 1 above)
Comment (upbqdn) [reply README.md:176]: How about the solution in [36d78dd](https://github.com/ZcashFoundation/zebra/pull/8654/commits/36d78ddbd65b7caf1d2da9c83cc2441e223b18c3)?

Comment (upbqdn) [reply README.md:176]: I can do something better if that doesn't suit.

Comment (arya2) [reply README.md:176]: That looks good!

**Review by arya2 — APPROVED** [review]

**Pull request merged**